library/src/test/java/com/novoda/downloadmanager/lib/BatchRepositoryTest.java

```diff
@@ -1,83 +1,363 @@
+        when(mockFileDownloadInfoId2.getBatchId()).thenReturn(2L);
+        when(mockFileDownloadInfoId3.getBatchId()).thenReturn(3L);
+        when(mockFileDownloadInfoId4.getBatchId()).thenReturn(4L);
     }
 
     @Test
     public void givenADownloadInfoWhenRetrievingTheBatchThenTheBatchIdsMatch() {
         long expectedBatchId = 100L;
-        when(fileDownloadInfo.getBatchId()).thenReturn(expectedBatchId);
-        Cursor batchCursor = singleBatchCursor(expectedBatchId);
-        when(contentResolver.query(any(Uri.class), any(String[].class), anyString(), any(String[].class), anyString())).thenReturn(batchCursor);
+        when(mockFileDownloadInfo.getBatchId()).thenReturn(expectedBatchId);
+        Cursor batchCursor = new MockCursorWithBatchIds(Arrays.asList(expectedBatchId));
+        when(mockContentResolver.query(any(Uri.class), any(String[].class), anyString(), any(String[].class), anyString())).thenReturn(batchCursor);
 
-        DownloadBatch downloadBatch = batchRepository.retrieveBatchFor(fileDownloadInfo);
+        DownloadBatch downloadBatch = batchRepository.retrieveBatchFor(mockFileDownloadInfo);
 
         assertThat(downloadBatch.getBatchId()).isEqualTo(expectedBatchId);
     }
 
-    private Cursor singleBatchCursor(final long batchId) {
-        final int batchSize = 1;
-        int idColumn = 1337;
-        Cursor cursor = mock(Cursor.class);
-        when(cursor.getColumnIndexOrThrow(Downloads.Impl.Batches._ID)).thenReturn(idColumn);
-        when(cursor.moveToNext()).thenAnswer(new Answer<Boolean>() {
-            int count = 0;
-
-            @Override
-            public Boolean answer(InvocationOnMock invocation) throws Throwable {
-                boolean batchSizeNotReached = count < batchSize;
-                count++;
-                return batchSizeNotReached;
-            }
-        });
-
-        when(cursor.getLong(idColumn)).thenReturn(batchId);
-
-        return cursor;
-    }
-
     @Test
     public void givenADownloadInfoAndNoLinkedBatchesWhenRetrievingTheBatchThenTheBatchIsDeleted() {
         long batchIdToBeMissing = 100L;
-        when(fileDownloadInfo.getBatchId()).thenReturn(batchIdToBeMissing);
+        when(mockFileDownloadInfo.getBatchId()).thenReturn(batchIdToBeMissing);
         Cursor emptyBatchCursor = mock(Cursor.class);
-        when(contentResolver.query(any(Uri.class), any(String[].class), anyString(), any(String[].class), anyString())).thenReturn(emptyBatchCursor);
+        when(mockContentResolver.query(any(Uri.class), any(String[].class), anyString(), any(String[].class), anyString())).thenReturn(emptyBatchCursor);
 
-        DownloadBatch downloadBatch = batchRepository.retrieveBatchFor(fileDownloadInfo);
+        DownloadBatch downloadBatch = batchRepository.retrieveBatchFor(mockFileDownloadInfo);
 
         assertThat(downloadBatch).isEqualTo(DownloadBatch.DELETED);
     }
+
+    @Test
+    public void whenThereAreFourBatchesMarkedToBeDeletedAndFourBatchesThenRemoveAll() {
```

Comment:
The test name is a little bit confusing. Should it be `givenThereAreFourBatchesMarkedToBeDeletedWhenDeletingMarkedBatchesForAllDownloadsThenItRemovesAll`?